File: Classes/Session/DTToolbarItems.m
//
//  DTToolbarItems.m
//  SimpleComic
//
//  Created by Alexander Rauchfuss on 7/18/09.
//  Copyright 2009 __MyCompanyName__. All rights reserved.
//

#import "DTToolbarItems.h"
#import "TSSTSessionWindowController.h"
#import "SimpleComicAppDelegate.h"


@implementation DTToolbarItem


-(void)validate
{
	TSSTSessionWindowController * toolbarDelegate = (TSSTSessionWindowController *)[[self toolbar] delegate];
	[(NSControl *)[self view] setEnabled: ![toolbarDelegate pageSelectionInProgress]];
}


@end


@implementation DTPageTurnToolbarItem


-(void)validate
{
	TSSTSessionWindowController * toolbarDelegate = (TSSTSessionWindowController *)[[self toolbar] delegate];

	[(NSSegmentedControl *)[self view] setEnabled: [toolbarDelegate canTurnPageLeft] forSegment: 0];
	[(NSSegmentedControl *)[self view] setEnabled: [toolbarDelegate canTurnPageRight] forSegment: 1];
	[super validate];
}


@end

File: Classes/Session/DTToolbarItems.swift
//
//  DTToolbarItems.swift
//  Simple Comic
//
//  Created by J-rg on 09.12.20.
//  Copyright © 2020 Dancing Tortoise Software. All rights reserved.
//

import Cocoa

class DTToolbarItem: NSToolbarItem {
	
	override func validate() {
		guard let toolbarDelegate = toolbar?.delegate as? TSSTSessionWindowController else { return }
		
		if toolbarDelegate.responds(to: #selector(getter: TSSTSessionWindowController.pageSelectionInProgress)) {
			(view as? NSControl)?.isEnabled = !(toolbarDelegate.pageSelectionInProgress)
		}
	}
}

class DTPageTurnToolbarItem: DTToolbarItem {
	
	override func validate() {
		guard let toolbarDelegate = toolbar?.delegate as? TSSTSessionWindowController else { return }
		
		if toolbarDelegate.responds(to: #selector(getter: TSSTSessionWindowController.canTurnPageLeft)) {
			(view as? NSSegmentedControl)?.setEnabled(toolbarDelegate.canTurnPageLeft, forSegment: 0)
		}
		
		if toolbarDelegate.responds(to: #selector(getter: TSSTSessionWindowController.canTurnPageRight)) {
			(view as? NSSegmentedControl)?.setEnabled(toolbarDelegate.canTurnPageRight, forSegment: 1)
		}
		
		super.validate()
	}
}
